qiskit/dagcircuit/dagcircuit.py

```diff
@@ -920,25 +920,31 @@ def node(self, node_id):
         Returns:
             node: the node.
         """
-        return self._multi_graph.nodes[node_id]
+        return self._multi_graph.get_node_data(node_id)
 
     def nodes(self):
         """Iterator for node values.
 
         Yield:
             node: the node.
         """
-        for node in self._multi_graph.nodes:
+        for node in self._multi_graph.nodes():
             yield node
 
     def edges(self, nodes=None):
-        """Iterator for node values.
+        """Iterator for edge values and source and dest node
 
         Yield:
-            node: the node.
+            edge: the edge.
         """
-        for source_node, dest_node, edge_data in self._multi_graph.edges(nodes, data=True):
-            yield source_node, dest_node, edge_data
+        if nodes is None:
+            nodes = self._multi_graph.nodes()
+
+        elif isinstance(nodes, DAGNode):
+            nodes = [nodes]
+        for node in nodes:
+            for source, dest, edge in self._multi_graph.in_edges(node._node_id):
```

Comment:
Agree we should update the docstring. On a valid `DAGCircuit`, for nodes of type `op`, `in_edges == out_edges`, but for `in`/`out` nodes, they would differ.

Reply:
I will switch this to `out_edges` so it's consistent with what we got before. But, I just tested it locally to figure out why it didn't cause any failures. It doesn't really matter because of the structure of the dagcircuit as @kdk pointed out, since input nodes have no parents and output nodes have no children we still return all the edges for the case when nodes is None. But it could matter when nodes are set. But, I only found one place in the code base where `edges()` is called with the node params set, in the commutation analysis pass where it runs:

```
for node in dag.topological_op_nodes:
    dag.edges(node)
```

so we never encounter that edge case in the tests. When I switch to `out_edges` I'll add a test case for it.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/3810/commits/5f46cc5a21c58be28a71b002b66a49f62d21dbda